R/jointRPCA.R

```diff
@@ -0,0 +1,752 @@
+#' Joint RPCA front-end and orchestration
+#'
+#' User-facing wrappers and internal orchestration of Joint Robust Principal Component Analysis (Joint-RPCA)
+#' for compositional multi-omic data using the OptSpace algorithm.
+#'
+#' This file contains:
+#' \itemize{
+#'   \item \code{jointRPCAuniversal()} — universal front-end detecting input type
+#'         (\code{matrix}, \code{list}, \code{SummarizedExperiment}, or \code{MultiAssayExperiment}).
+#'   \item \code{runJointRPCA()} — high-level wrapper that runs Joint-RPCA and
+#'         stores the embedding in \code{reducedDim()}.
+#'   \item \code{.jointRPCA()} — internal Joint-RPCA engine handling preprocessing,
+#'         shared-sample alignment, factorization, and output assembly.
+#'   \item \code{.joint_optspace_helper()} — internal coordination of multi-view OptSpace factorization.
+#'   \item \code{.transform()} and \code{.transform_helper()} — projection of new data into existing ordination space.
+#'   \item \code{.rpca_table_processing()} — compositional preprocessing (filtering, zero-removal, etc.).
+#'   \item \code{.mask_value_only()} — safe numeric matrix masking for missing values.
+#'   \item Other small utility functions used across the Joint-RPCA workflow.
+#' }
+#'
+#' These functions depend on the OptSpace backend defined in \code{optspace.R}.
+#' Only \code{jointRPCAuniversal()} and \code{runJointRPCA()} are exported; all others are internal helpers.
+#'
+#' @keywords internal
+#' @noRd
+NULL
+
+#' Universal Joint RPCA Wrapper
+#'
+#' @param x Input object: \code{MultiAssayExperiment}, \code{SummarizedExperiment}
+#'   (including \code{TreeSummarizedExperiment}), list of matrices, or single matrix.
+#' @param experiments Character vector of experiment names to extract when \code{x}
+#'   is a \code{MultiAssayExperiment} (i.e. \code{names(experiments(x))}).
+#'   If \code{NULL}, all experiments are used.
+#'
+#'   For \code{MultiAssayExperiment} inputs, \strong{one assay per experiment} is
+#'   used: by default the first assay returned by
+#'   \code{SummarizedExperiment::assayNames()} (or index \code{1L} if unnamed).
+#'   The actually used assay names are recorded in \code{$assay.names.used} in
+#'   the result. If you need a different assay (e.g. \code{"relab"} instead of
+#'   \code{"counts"}), subset or reorder assays in \code{x} before calling
+#'   \code{jointRPCAuniversal()}.
+#' @param ... Additional arguments passed to \code{.jointRPCA()}.
+#'
+#' @return Output from \code{.jointRPCA()} with extra fields when \code{x} is a
+#'   \code{MultiAssayExperiment}:
+#'   \itemize{
+#'     \item \code{$experiment.names}: character vector of experiments used.
+#'     \item \code{$assay.names.used}: named character vector giving, for each
+#'           experiment, the assay name that was used (typically the first in
+#'           \code{assayNames()}).
+#'   }
+#' @export
+
+jointRPCAuniversal <- function(x, experiments = NULL, ...) {
```

Comment:
You could use generic methods. One for SE, one for MAE. It is tidier way compared to checking the class inside the function.